### DISCUSSION TITLE
What's the right way to modify a default autocommand?

### DISCUSSION THREAD

**Discussion by a-torgovitsky** [LazyVim/LazyVim#3994] [Q&A]
This autocommand is loaded by LazyVim by default in `/lazyvim/config/autocmds.lua`:
```lua
-- wrap and check for spell in text filetypes
vim.api.nvim_create_autocmd("FileType", {
  group = augroup("wrap_spell"),
  pattern = { "text", "plaintex", "typst", "gitcommit", "markdown" },
  callback = function()
    vim.opt_local.wrap = true
    vim.opt_local.spell = true
  end,
})
```

I want to modify it so that it includes `tex` filetypes as well.
So I put this in my own `/config/autocmds.lua`:

```lua
-- This is defined at the top of /lazyvim/config/autocmds.lua
local function augroup(name)
  return vim.api.nvim_create_augroup("lazyvim_" .. name, { clear = true })
end

vim.api.nvim_create_autocmd("FileType", {
  group = augroup("wrap_spell"),
  pattern = { "tex", "text", "plaintex", "typst", "gitcommit", "markdown" },
  callback = function()
    vim.opt_local.wrap = true
    vim.opt_local.spell = true
  end,
})
```

That achieves what I want, but is it the preferred way to do this with LazyVim?
It feels a bit hacky compared to the approaches for changing plugin defaults, for example.
I suppose my concern is that I am overwriting the entire autocommand rather than just amending `tex` to the set of patterns, which seems like it would be more consistent with how plugin configuration works with LazyVim.

**Comment by pidgeon777** [comment]
I'm also curious to know this. Also, how to remove a default `LazyVim` autocommand?

**Comment by louwers** [comment]
This needs to be documented!

The default autocmds also need to be documented.

I want to disable spell check, `vim.opt.spell = false` didn't cut it. Then I discovered there is an autocmd that sets this up...

```lua
vim.api.nvim_del_augroup_by_name("lazyvim_wrap_spell")
```

**Reply by awilkins** [reply]
Thank you so much for this hint, I was nearly there but I hadn't grokked the `augroup()` function at the top of the file.

**Comment by Raaghav-Arya** [comment]
Adding a counter autcmd like below in my lua/config/autocmds.lua file worked:

```lua
-- Disable wrap for markdown files (override LazyVim default)
vim.api.nvim_create_autocmd("FileType", {
  group = vim.api.nvim_create_augroup("disable_markdown_wrap", { clear = true }),
  pattern = "markdown",
  callback = function()
    vim.opt_local.wrap = false
  end,
})
```
